docs/content/preview/troubleshoot/nodes/disk-full.md

```diff
@@ -1,25 +1,136 @@
+
+During YB-TServer startup, the FSManager (File System Manager) module checks the health of each data drive by creating a small test file. If the disk is completely full, the YB-TServer does the following:
+
+1. Skips the full drive: The test file creation fails, the node ignores this particular data drive, and marks it faulty. It also increments the `drive_fault` metric.
+
+1. Reports faulty drive: YB-TServer informs YB-Master about the faulty drive issue. The YB-Master server then blacklists the YB-TServer with the faulty drive, preventing it from creating any new tablets.
+
+1. Potential tablet migration: The load balancer might load balance tablets from the blacklisted YB-TServer to other YB-TServers for {{<release "2.20.3.0">}} or {{<release "2024.1.0.0">}} and later. For earlier releases, the YB-TServer can run into the following fatal error:
+
+    ```output
+    Log line format: [IWEF]mmdd hh:mm:ss.uuuuuu threadid file:line] msg
+    F0119 04:08:47.225858 46656 tablet_server_main_impl.cc:241] Already present (yb/tserver/ts_tablet_manager.cc:1554): Could not init Tablet Manager: State transition of tablet d1e9c6c53fa9410c86d3505073305b41 already in progress:     {d1e9c6c53fa9410c86d3505073305b41, opening tablet}
+    ```
+
+## Recommended recovery actions
+
+The following two recommendations are specific to [YugabyteDB Anywhere](../../../yugabyte-platform/).
+
+**Option 1**: Increase the disk volume size using the [Edit Universe](../../../yugabyte-platform/manage-deployments/edit-universe/#edit-a-universe) option. Note the following important considerations:
+
+- Preflight checks - YugabyteDB Anywhere performs preflight checks to ensure a smooth [smart resize](../../../yugabyte-platform/manage-deployments/edit-universe/#smart-resize) operation. One of the checks verifies there is no leaderless tablet in the cluster. To bypass this check, you can disable the `yb.checks.leaderless_tablets.enabled` global [runtime configuration flag](../../../yugabyte-platform/administer-yugabyte-platform/manage-runtime-config/) as a Super Admin user.
+
+- Master leader - The universe needs to have a designated master leader node to receive and process node resize requests. If no master leader is available, the resize operation will fail. After the disk space is successfully increased, the bootstrap process for the resized nodes should proceed without issues.
+
+**Option 2**: Add a new node using the [Edit Universe](../../../yugabyte-platform/manage-deployments/edit-universe/#edit-a-universe) option.
+
+Note that in a cluster with a [replication factor](../../../architecture/docdb-replication/replication/#replication-factor) (RF) of N, if more than N/2 nodes fail, adding a new node will fail. To recover from this situation, you need to bring some failed nodes back online (using Option 1 or Option 2) to form a quorum, and then add new nodes. If that's not possible, you might need to manually perform an unsafe configuration change.
```

Comment:
"bring some failed nodes back online (using Option 1 or Option 2)" - option 2 is add a new node, but in this situation "adding a new node will fail"

Reply:
"unsafe configuration change" - what is that? like what?